src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs

```diff
@@ -7,27 +7,76 @@ namespace Stripe
     using Newtonsoft.Json;
     using Stripe.Infrastructure;
 
+    /// <summary>
+    /// Global configuration class for Stripe.net settings.
+    /// </summary>
     public static class StripeConfiguration
     {
-        private static string stripeApiVersion = "2018-11-08";
-
         private static string apiKey;
-        private static string apiBase;
-        private static string connectBase;
-        private static string filesBase;
-        private static JsonSerializerSettings serializerSettings = DefaultSerializerSettings();
 
         static StripeConfiguration()
         {
             StripeNetVersion = new AssemblyName(typeof(Requestor).GetTypeInfo().Assembly.FullName).Version.ToString(3);
         }
 
-        public static string StripeApiVersion
+        /// <summary>Default base URL for Stripe's API.</summary>
+        public static string DefaultApiBase => "https://api.stripe.com";
+
+        /// <summary>Default base URL for Stripe's OAuth API.</summary>
+        public static string DefaultConnectBase => "https://connect.stripe.com";
+
+        /// <summary>Default base URL for Stripe's Files API.</summary>
+        public static string DefaultFilesBase => "https://files.stripe.com";
+
+        /// <summary>Default timespan before the request times out.</summary>
+        public static TimeSpan DefaultHttpTimeout => new TimeSpan(80 * TimeSpan.TicksPerSecond);
+
+        /// <summary>Default API version used by the library.</summary>
+        public static string DefaultStripeApiVersion => "2018-11-08";
+
+        /// <summary>Gets or sets the base URL for Stripe's API.</summary>
+        public static string ApiBase { get; set; } = DefaultApiBase;
+
+#if NET45 || NETSTANDARD2_0
+        /// <summary>Gets or sets the API key.</summary>
+        /// <remarks>
+        /// You can also set the API key using the <c>StripeApiKey</c> key in
+        /// <see cref="System.Configuration.ConfigurationManager.AppSettings"/>.
+        /// </remarks>
```

Comment:
I feel weird about preprocessor commands + comments. Isn't it better to document it globally?

Reply:
The issue is that we can't link to `System.Configuration.ConfigurationManager.AppSettings` with .NET Standard 1.2 because the class doesn't exist. I could just remove the link or the comment entirely, but it's kind of a shame.

Reply:
Ah yeah I see what you mean. Ugh. Sounds good then!